### DISCUSSION TITLE
Error message on UGS on Jog issued from controller

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#516] [Config and issues] [answered]
(Using uCNC 1.7.4 with Mega/Ramps 1.4 and Reprapdiscount full graphic smart controller)

When the controller sends a Jog command and is connected to UGS, UGS displays this error message:

"Error while processing response <ok>: An unexpected command was completed by the controller." 

When the controller command is "Home", UGS displays two of this messages in sequence, but only in the last homing step (Homing switch pull-off distance).

This is not really a problem, but maybe makes a cleaner user experience.

**Comment by Paciente8159** [comment]
Using an app or the display controller there is no real distinction. All command requests are all pipelined on to a single buffer and all responses are broadcasted to all communications ports.
Probably the same will happen if you run 2 instances of UGS in different uart ports or usb or wifi.
UGS just expects to be the only one to control the board and issues errors on replies that were not preceded by a command.

There is no real way to overcome this. Isolating each com port would make the firmware much more complex.

**Comment by Paciente8159** [comment]
After dwelling for while on this and having faced this same necessity with a new idea I had (and am currently working on) I am going to pick up this as a request to channel the communications correctly.

I'll post some feedback on this.

**Comment by Paciente8159** [comment]
FR #522 created

**Comment by Paciente8159** [comment]
This [branch](https://github.com/Paciente8159/uCNC/tree/522-fr-communication-revision-multi-controllers) has the proposed changes to work with multistream. This is now going to enter tests phase.

This is just the code changes necessary to support multistream. The display module needs to be adapted to make use of multistreams, This way all commands sent by the display will not have any interference in the UGS or other software connections.

**Comment by Paciente8159** [comment]
And it's [here](https://github.com/Paciente8159/uCNC/pull/529).

It's ready to be integrated. With it also some modules will be affected to take advantage of the new multistream capabilities (mainly graphic display and sd card modules)

**Comment by Paciente8159** [comment, marked as answer]
This should be fixed now with the latest [v1.8.0](https://github.com/Paciente8159/uCNC/releases/tag/v1.8.0)

**Reply by mvackel** [reply]
I've just tested. Works fine! Clean and nice. Great job!
